lib/flagged_revs_filter.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+
+const P = require('bluebird');
+const URI = require('hyperswitch').URI;
+const mwUtil = require('./mwUtil');
+
+
+module.exports = (hyper, req, next, options, specInfo) => {
+
+    const rp = req.params;
+
+    if (mwUtil.isNoCacheRequest(req) || rp.revision || rp.revision === 0) {
```

Comment:
What's up with a zero? Is it possible to have a 0 revision number?

Reply:
`0` is used for stashing, and since it's false-y, it needs special treatment here.

Reply:
In stashing code we `'0'` is added afterwards inside the parsed module. https://github.com/wikimedia/restbase/blob/master/sys/parsoid.js#L769

So at this level 0 shouldn't be tolerated.

Reply:
This check will not be exercised as we don't apply the filter on POSTs nor on requests containing a revision inside the spec, but it's future-proof. AFAIK, once we add the revision `0`, that is picked up by VE and used on subsequent requests, so I'd prefer to keep it there. Invalid or bogus revision handling should be left to the Parsoid module either way, we should just ensure that the filter does the right thing.

Reply:
I was nit picking :) Agreed. It doesn't hurt to have the check.